source/fuzz/transformation_flatten_conditional_branch.cpp

```diff
@@ -0,0 +1,600 @@
+    for (uint32_t id : pair.id()) {
+      fresh_ids.push_back(id);
+    }
+
+    auto instruction =
+        FindInstruction(pair.instruction_descriptor(), ir_context);
+    if (instruction) {
+      instructions_to_fresh_ids.emplace(instruction, fresh_ids);
+    }
+  }
+
+  return instructions_to_fresh_ids;
+}
+
+uint32_t TransformationFlattenConditionalBranch::NumOfFreshIdsNeededByOpcode(
+    SpvOp opcode) const {
+  switch (opcode) {
+    case SpvOpStore:
+      return 2;
+    case SpvOpLoad:
+    case SpvOpFunctionCall:
+      return 5;
+    default:
+      assert(false && "This line should never be reached");
+      return 0;
+  }
+}
+
+opt::BasicBlock*
+TransformationFlattenConditionalBranch::EncloseInstructionInConditional(
+    opt::IRContext* ir_context, TransformationContext* transformation_context,
+    opt::BasicBlock* block, opt::Instruction* instruction,
+    std::vector<uint32_t> fresh_ids, uint32_t condition_id,
+    bool exec_if_cond_true) const {
+  assert((instruction->opcode() == SpvOpStore ||
+          instruction->opcode() == SpvOpLoad ||
+          instruction->opcode() == SpvOpFunctionCall) &&
+         "This function should only be called on OpStore, OpLoad or "
+         "OpFunctionCall instructions.");
+
+  // Get the next instruction (it will be useful for splitting).
+  auto next_instruction = instruction->NextNode();
+
+  // We need at least 2 fresh ids for two new blocks.
+  assert(fresh_ids.size() >= 2 && "Not enough fresh ids.");
+
+  // Update the module id bound
+  for (auto id : fresh_ids) {
+    fuzzerutil::UpdateModuleIdBound(ir_context, id);
+  }
+
+  // Create the block where the instruction is executed by splitting the
+  // original block.
+  auto execute_block = block->SplitBasicBlock(
+      ir_context, fresh_ids[0],
+      fuzzerutil::GetIteratorForInstruction(block, instruction));
+
+  // Create the merge block for the conditional that we are about to create by
+  // splitting execute_block (this will leave |instruction| as the only
+  // instruction in |execute_block|).
+  auto merge_block = execute_block->SplitBasicBlock(
+      ir_context, fresh_ids[1],
+      fuzzerutil::GetIteratorForInstruction(execute_block, next_instruction));
+
+  // Propagate the fact that the block is dead to the newly-created blocks.
+  if (transformation_context->GetFactManager()->BlockIsDead(block->id())) {
+    transformation_context->GetFactManager()->AddFactBlockIsDead(
+        execute_block->id());
+    transformation_context->GetFactManager()->AddFactBlockIsDead(
+        merge_block->id());
+  }
+
+  // Initially, consider the merge block as the alternative block to branch to
+  // if the instruction should not be executed.
+  auto alternative_block = merge_block;
+
+  // Add an unconditional branch from |execute_block| to |merge_block|.
+  execute_block->AddInstruction(MakeUnique<opt::Instruction>(
+      ir_context, SpvOpBranch, 0, 0,
+      std::initializer_list<opt::Operand>{opt::Operand(
```

Comment:
Consider using `opt::Instruction::OperandList` instead of `std::initializer_list<opt::Operand>`. If you do so, you can get rid of `opt::Operand` and replace it with `{}`.

Reply:
Done